frontend/src/compononents/CocktailRecipe.tsx

```diff
@@ -2,41 +2,40 @@ import './CocktailRecipe.css';
 import {Cocktail} from "../model/Cocktail";
 import * as React from "react";
 import {Button} from "@mui/material";
-
+import {Favourite} from "../model/Favourite";
 
 type CocktailRecipeProps = {
-    cocktail : Cocktail
+    cocktail: Cocktail
+    addFavourite: (favourite: Favourite) => void
 }
 
-export default function CocktailRecipe( { cocktail } : CocktailRecipeProps ){
-
+export default function CocktailRecipe( { cocktail,addFavourite} : CocktailRecipeProps ){
 
     return <div className="cocktail-recipe_" >
         <div className="cocktail-recipe_card" >
-
         <div className="wrapper">
-            <div className="box a">{cocktail.name}<Button variant="text">Add to favorites</Button></div>
+            <div className="box a">{cocktail.name}<Button variant="text" onClick={()=> addFavourite(cocktail)}>Add to favorites</Button></div>
```

Comment:
Ich habe bemerkt, dass in dieser Klasse die Einrückung noch nicht passt. Dies ist eigentlich nicht Teil des Pull-Requests, aber meine Empfehlung ist IntelliJ für das Formatieren zu nutzen.